js/src/views/ParityBar/parityBar.js

```diff
@@ -45,12 +46,31 @@ class ParityBar extends Component {
     }
 
     if (count < newCount) {
-      this.setState({ opened: true });
+      this.setOpened(true);
     } else if (newCount === 0 && count === 1) {
-      this.setState({ opened: false });
+      this.setOpened(false);
     }
   }
 
+  setOpened (opened) {
+    this.setState({ opened });
+
+    if (!this.bar) {
+      return;
+    }
+
+    // Fire up custom even to support having parity bar iframed.
+    const event = new CustomEvent('parity.bar.visibility', {
+      bubbles: true,
+      detail: { opened }
+    });
+    this.bar.dispatchEvent(event);
+  }
+
+  onRef = (el) => {
```

Comment:
Would have preferred slightly more descriptive variable naming, i.e. `element` (we gain understanding reading, lose nothing after compilation)